elastic/integrations
# [Platform Observability] Create initial PO package for ingesting kibana ECS formatted logs

## What does this PR do?

This PR is the initial work for creating the new Platform Observability package. We will start with ingesting Kibana logs but its capabilities will be extended over time. This package is heavily based on the solution provided here https://github.com/elastic/integrations/pull/2972

## Checklist


## Author's Checklist

- [x] Define the package name 
- [x] Align current Kibana package ingest pipelines with this one  (This is a duplication of the existing Kibana package)
- [x] Decide whether we will go with Filebeat processors to parse the json object or not

## How to test this PR locally
In  order to tests this we first need to edit the configuration used by `elastic-packages`

* Enable kibana logging, by adding the configuration below to ` ~/.elastic-package/profiles/default/stack/kibana.config.8x.yml`
```
xpack.security.audit.enabled: true
xpack.security.audit.appender:
  type: rolling-file
  fileName: ./.kbn-logs/audit.log
  policy:
    type: time-interval
    interval: 24h
  strategy:
    type: numeric
    max: 10
  layout:
    type: json

logging:
  appenders:
    file:
      type: file
      fileName: ./.kbn-logs/kibana.log
      layout:
        type: json
  root:
    appenders: [default, file]
    level: all

  loggers:
    - name: server
      level: fatal
    - name: telemetry
      appenders: [file]
      level: all
    - name: metrics.ops
      appenders: [file]
      level: debug
```
* Edit ` ~/.elastic-package/profiles/default/stack/snapshot.yml`, adding `"../../../tmp/service_logs:/usr/share/kibana/.kbn-logs"` to the list of volumes for `kibana` container, to enable the agent to see the logs
* From the repo root folder go to `packages/platform_observability` folder and run `elastic-package build`
* Run `elastic-package stack up -v -d`
* On Kibana, go to `Integrations` 
* Install `Platform Observability` with default configuration, adding it to `Elastic-Agent` host

## Related issues

- Closes [#2226](https://github.com/elastic/observability-dev/issues/2226)

## Screenshots
**Integration page**
<img width="1287" alt="image" src="https://user-images.githubusercontent.com/2767137/179688807-5acc94b1-26e7-405a-8139-5b3f89e24a2b.png">

**Kibana logs**
<img width="1443" alt="image" src="https://user-images.githubusercontent.com/2767137/176436589-43a042d2-77e4-499d-b6ad-caa73db7afa0.png">

**Kibana audit logs**
<img width="1475" alt="image" src="https://user-images.githubusercontent.com/2767137/176436440-562a4bac-b529-4472-8ce1-d915d3161242.png">
